File: BATabBarPOC/BATabBarPOC/BATabBarPOC/ViewController.m
//
//  ViewController.m
//  BATabBarPOC
//
//  Created by Qingchuan Zhu on 11/28/17.
//  Copyright © 2017 Qingchuan Zhu. All rights reserved.
//

#import "ViewController.h"
#import "BATabBarController.h"
#import "BottomViewController1.h"
#import "BottomViewController2.h"
#import "TopViewController.h"
#import "TopViewController_a2.h"
#import "BATabBarController_a1.h"
#import "BATabBarController_a2.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UISwitch *pinnedButtonCtl;
@property (nonatomic, assign) BOOL shouldShowPinnedButton;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    self.pinnedButtonCtl.on = NO;
}

- (IBAction)switchChanged:(UISwitch *)sender {
    if (sender.on) {
        self.shouldShowPinnedButton = YES;
    } else {
        self.shouldShowPinnedButton = NO;
    }
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)testForNewWay:(id)sender {
    BATabBarController_a1 *tabBar2 = [BATabBarController_a1 new];
    TopViewController_a2 *topVC = [TopViewController_a2 new];
    BottomViewController1 *vc1 = [BottomViewController1 new];
    BottomViewController2 *vc2 = [BottomViewController2 new];
    vc1.tabBarDelegate = tabBar2;
    vc2.tabBarDelegate = tabBar2;
    tabBar2.topViewController = topVC;
    tabBar2.childViewControllers = (NSMutableArray<BARRTabBarChildProtocol>*)[NSMutableArray arrayWithArray:@[vc1, vc2]];
    tabBar2.showPinnedButton = self.shouldShowPinnedButton;
    [self.navigationController pushViewController:tabBar2 animated:YES];
}

- (IBAction)testForAlpha2Way:(id)sender {
    BATabBarController_a2 *tabBar3 = [BATabBarController_a2 new];
    TopViewController *topVC = [TopViewController new];
    BottomViewController1 *vc1 = [BottomViewController1 new];
    BottomViewController2 *vc2 = [BottomViewController2 new];
    vc1.tabBarDelegate = tabBar3;
    vc2.tabBarDelegate = tabBar3;
    tabBar3.topViewController = topVC;
    tabBar3.childViewControllers = (NSMutableArray<BARRTabBarChildProtocol>*)[NSMutableArray arrayWithArray:@[vc1, vc2]];
    tabBar3.showPinnedButton = self.shouldShowPinnedButton;
    [self.navigationController pushViewController:tabBar3 animated:YES];
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender{
    if ([[segue destinationViewController] isKindOfClass:[BATabBarController class]]) {
        BATabBarController *destVC = (BATabBarController *)[segue destinationViewController];
        BottomViewController1 *vc1 = [BottomViewController1 new];
        BottomViewController2 *vc2 = [BottomViewController2 new];
        destVC.childViewControllers = @[vc1, vc2];
    }
}

@end

File: AnchorPointTest/AnchorPointTest/ViewController.swift
//
//  ViewController.swift
//  AnchorPointTest
//
//  Created by Qingchuan Zhu on 7/19/18.
//  Copyright © 2018 Qingchuan Zhu. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet var testingLabel_default: UILabel!
    @IBOutlet var button1: UIButton!
    @IBOutlet var anchorLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        self.anchorLabel.layer.anchorPoint = CGPoint(x: 1, y: 0.5)
        self.anchorLabel.bounds.origin = CGPoint(x: 0, y: anchorLabel.bounds.size.width)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func incrementWidth(_ sender: UIButton) {
        var newFrame:CGRect = self.testingLabel_default.frame
        newFrame.size.width +=  30
        self.testingLabel_default.frame = newFrame
    }
    
    @IBAction func incrementAnchorLabel(_ sender: Any) {
        var newFrame:CGRect = self.anchorLabel.frame
        newFrame.size.width +=  30
        self.anchorLabel.frame = newFrame
    }
}

